src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs

```diff
@@ -1334,6 +1336,95 @@ select Task.Factory.StartNew(() => pair.Item1.Receive(new byte[1]), Cancellation
         }
     }
 
+    public class BlockingSendReceive
+    {
+
+        private readonly ITestOutputHelper _log;
+        public static readonly object[][] Loopbacks = MemberDatas.Loopbacks;
+
+        public BlockingSendReceive(ITestOutputHelper output)
+        {
+            _log = output;
+        }
+
+        [Theory]
+        [MemberData(nameof(Loopbacks))]
+        public async Task BlockingStreamRead_Close(IPAddress ipAddress)
+        {
+            // Test for #22564 when close is called while blocking read is pending.
+            using (var server = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            using (var client = new Socket(ipAddress.AddressFamily, SocketType.Stream, ProtocolType.Tcp))
+            {
+                server.BindToAnonymousPort(ipAddress);
+                server.Listen(1);
+                byte[] buffer = new byte[1];
+                buffer[0] = Convert.ToByte('a');
+
+                Task clientTask = Task.Run(() =>
+                {
+                    client.Connect(server.LocalEndPoint);
+                    client.Send(buffer);
+                    // Blocking read.
+                    try
+                    {
+                        client.Receive(buffer);
```

Comment:
What is the expected behavior here when we Close? Do we throw? Do we return 0 bytes? Seems like we should validate the expected behavior.

Reply:
It will throw one or the other exception in catch. It it throws something else or if it hangs, test should fail.